pkg/handlers/apk.go

```diff
@@ -0,0 +1,363 @@
+		ctx,
+		"filename", fileName,
+	)
+	return h.handleNonArchiveContent(ctx, mimeReader, apkChan)
+}
+
+// createZipReader creates a new ZIP reader from the input fileReader.
+func createZipReader(input fileReader) (*zip.Reader, error) {
+	size, err := input.Seek(0, io.SeekEnd)
+	if err != nil {
+		return nil, err
+	}
+	// Reset the reader position to the start
+	_, err = input.Seek(0, io.SeekStart)
+	if err != nil {
+		return nil, err
+	}
+	// Create a new ZIP reader for the data
+	zipReader, err := zip.NewReader(input, size)
+	if err != nil {
+		return nil, err
+	}
+	return zipReader, err
+}
+
+// parseResTable parses the resources.arsc file and returns the ResourceTable.
+func parseResTable(zipReader *zip.Reader) (*apkparser.ResourceTable, error) {
+	for _, file := range zipReader.File {
+		if file.Name == "resources.arsc" {
+			rdr, err := readFile(file)
+			if err != nil {
+				return nil, err
+			}
+			defer rdr.Close()
+
+			resTable, err := apkparser.ParseResourceTable(rdr)
+			if err != nil {
+				return nil, err
+			}
+			return resTable, nil
+		}
+	}
+	return nil, errors.New("resources.arsc file not found in the APK archive")
+}
+
+// readFile reads the file from the zip archive and returns the data as an io.ReadCloser
+// Note: responsibility of calling function to close the reader
+func readFile(file *zip.File) (io.ReadCloser, error) {
+	rc, err := file.Open()
+	if err != nil {
+		return nil, err
+	}
+	return rc, nil
+}
+
+// hasSubstring checks if the string contains any of the provided substrings.
+func hasSubstring(s string, substrings []string) bool {
+	for _, sub := range substrings {
+		if strings.Contains(s, sub) {
+			return true
+		}
+	}
+	return false
+}
+
+// extractStringsFromResTable extracts the strings from the resources table
+// Note: This is a hacky way to get the strings from the resources table
+// APK strings are typically (always?) stored in the 0x7f000000-0x7fffffff range
+// https://chromium.googlesource.com/chromium/src/+/master/build/android/docs/life_of_a_resource.md
+func extractStringsFromResTable(resTable *apkparser.ResourceTable) (io.Reader, error) {
+	var resourceStrings bytes.Buffer
+	inStrings := false
+	for i := 0x7f000000; i <= 0x7fffffff; i++ {
+		entry, _ := resTable.GetResourceEntry(uint32(i))
+		if entry == nil {
+			continue
+		}
+		if entry.ResourceType == "string" {
+			inStrings = true
+			val, err := entry.GetValue().String()
+			if err != nil {
+				return nil, err
+			}
+			// Write directly to the buffer
+			_, err = resourceStrings.WriteString(fmt.Sprintf("%s: %s\n", entry.Key, val))
+			if err != nil {
+				return nil, err
+			}
+		}
+		// Exit the loop if we've finished processing the strings
+		if inStrings && entry.ResourceType != "string" {
+			break
+		}
+	}
+	return &resourceStrings, nil
+}
+
+// processDexFile decodes the dex file and returns the relevant instructions
+func processDexFile(ctx logContext.Context, rdr io.ReadCloser) (io.Reader, error) {
```

Comment:
It would be nice to have this outside of the apk specific case like if we find `.dex` file directly outside

Reply:
Great idea. Might be something for a separate PR, just so that this can get out the door. But I like where you're going with this. I think we should probably have several additional file handlers to handle specific types like `dex`, `pyc`, etc.